src/main/java/com/cleanroommc/groovyscript/compat/mods/extrautils2/Crusher.java

```diff
@@ -0,0 +1,160 @@
+package com.cleanroommc.groovyscript.compat.mods.extrautils2;
+
+import com.cleanroommc.groovyscript.api.GroovyLog;
+import com.cleanroommc.groovyscript.api.IIngredient;
+import com.cleanroommc.groovyscript.api.documentation.annotations.*;
+import com.cleanroommc.groovyscript.compat.mods.ModSupport;
+import com.cleanroommc.groovyscript.helper.SimpleObjectStream;
+import com.cleanroommc.groovyscript.helper.ingredient.IngredientHelper;
+import com.cleanroommc.groovyscript.helper.recipe.AbstractRecipeBuilder;
+import com.cleanroommc.groovyscript.registry.VirtualizedRegistry;
+import com.rwtema.extrautils2.api.machine.IMachineRecipe;
+import com.rwtema.extrautils2.api.machine.MachineSlotFluid;
+import com.rwtema.extrautils2.api.machine.MachineSlotItem;
+import com.rwtema.extrautils2.api.machine.XUMachineCrusher;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fluids.FluidStack;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+@RegistryDescription
+public class Crusher extends VirtualizedRegistry<IMachineRecipe> {
+
+    @Override
+    public void onReload() {
+        removeScripted().forEach(XUMachineCrusher.INSTANCE.recipes_registry::removeRecipe);
+        restoreFromBackup().forEach(XUMachineCrusher.INSTANCE.recipes_registry::addRecipe);
+    }
+
+    public IMachineRecipe add(IMachineRecipe recipe) {
+        if (recipe != null) {
+            addScripted(recipe);
+            XUMachineCrusher.INSTANCE.recipes_registry.addRecipe(recipe);
+        }
+        return recipe;
+    }
+
+    public boolean remove(IMachineRecipe recipe) {
+        if (XUMachineCrusher.INSTANCE.recipes_registry.removeRecipe(recipe)) {
+            addBackup(recipe);
+            return true;
+        }
+        return false;
+    }
+
+    @MethodDescription(description = "groovyscript.wiki.removeByInput", example = @Example("item('minecraft:blaze_rod')"))
+    public void removeByInput(IIngredient input) {
+        List<IMachineRecipe> agony = new ArrayList<>();
+        for (IMachineRecipe recipe : XUMachineCrusher.INSTANCE.recipes_registry) {
+            for (Pair<Map<MachineSlotItem, List<ItemStack>>, Map<MachineSlotFluid, List<FluidStack>>> mapMapPair : recipe.getJEIInputItemExamples()) {
+                for (ItemStack stack : mapMapPair.getKey().get(XUMachineCrusher.INPUT)) {
+                    if (input.test(stack)) {
+                        agony.add(recipe);
+                    }
+                }
+            }
+        }
+        for (IMachineRecipe recipe : agony) {
+            addBackup(recipe);
+            XUMachineCrusher.INSTANCE.recipes_registry.removeRecipe(recipe);
+        }
+    }
```

Comment:
I guess we no longer log invalid outputs and if no recipes were found. Thats fine, but we should at least return a boolean if any recipes where removed.